### PR TITLE
CDN notify scheduler task has been cleaned

### PR DESCRIPTION
**Pull request by 244372610**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] Code compiles correctly.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: cdn/cdn.go [hunk 1]
```diff
@@ -52,32 +56,63 @@ type Server struct {
 	// Manager client
 	configServer managerClient.Client
 
+	// dynamic config
+	dynconfig dynconfig.Interface
```
Comment (yxxhero) [inline cdn/cdn.go:60]: var dynconfig has same name for dynconfig package?

File: cdn/cdn.go [hunk 2]
```diff
@@ -104,31 +139,56 @@ func New(config *config.Config) (*Server, error) {
...
+			})
+			if err != nil {
+				return nil, err
+			}
+			return cdn, nil
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "create dynamic config")
+		}
+	} else {
```
Comment (yxxhero) [inline cdn/cdn.go:176]: remove else if better?

File: cdn/dynconfig/config.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	RefreshInterval time.Duration `yaml:"refreshInterval" mapstructure:"refreshInterval"`
+
+	// CachePath is cache file path.
+	CachePath string `yaml:"cachePath" mapstructure:"cachePath"`
+
+	// sourceType is source type of dynConfig,
+	SourceType dc.SourceType `yaml:"sourceType" mapstructure:"sourceType"`
+
+	//// ConfigPath is dynamic config path when sourceType is local
+	//ConfigPath string `yaml:"configPath" mapstructure:"configPath"`
```
Comment (yxxhero) [inline cdn/dynconfig/config.go:39]: ////?

File: cdn/dynconfig/config.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+func (c Config) Validate() []error {
+	var errors []error
+	if c.SourceType != dc.ManagerSourceType && c.SourceType != dc.LocalSourceType {
+		errors = append(errors, fmt.Errorf("dynamic config sourceType only support manager or local but current is %s",
+			c.SourceType))
+	}
+	//if c.SourceType == dc.LocalSourceType && c.ConfigPath == "" {
+	//	errors = append(errors, fmt.Errorf("dynamic config sourceType is local but dynamic config path is empty"))
+	//}
```
Comment (yxxhero) [inline cdn/dynconfig/config.go:50]: remove

File: cdn/rpcserver/rpcserver.go [hunk 5]
```diff
@@ -244,10 +248,73 @@ func (css *Server) GetPieceTasks(ctx context.Context, req *base.PieceTaskRequest
 	return pp, nil
 }
 
+//func (css *Server) SyncPieceTasks(stream cdnsystem.Seeder_SyncPieceTasksServer) error {
```
Comment (yxxhero) [inline cdn/rpcserver/rpcserver.go:251]: remove

File: cdn/rpcserver/rpcserver.go [hunk 6]
```diff
@@ -103,6 +103,10 @@ func (css *Server) ObtainSeeds(ctx context.Context, req *cdnsystem.SeedRequest,
 		span.RecordError(err)
 		return err
 	}
+	//css.service.AddTaskGCSubscriber(req.TaskId, &task.GCSubscriberInstance{
```
Comment (yxxhero) [inline cdn/rpcserver/rpcserver.go:106]: remove

Comment (244372610) [reply cdn/rpcserver/rpcserver.go:106]: PR has not been completed, now it is just testing, and the comments will be deleted later, for the same reason above

File: .github/workflows/lint.yml [hunk 7]
```diff
@@ -21,7 +21,7 @@ jobs:
       - name: Golangci lint
         uses: golangci/golangci-lint-action@v2
         with:
-          version: v1.40.1
+          version: latest
```
Comment (gaius-qi) [inline .github/workflows/lint.yml:24]: Don't change lint version.

File: internal/dynconfig/dynconfig.go [hunk 8]
```diff
@@ -156,6 +158,17 @@ func (d *Dynconfig) Unmarshal(rawVal interface{}) error {
 	return d.strategy.Unmarshal(rawVal)
 }
 
+func (d *Dynconfig) Get() (interface{}, error) {
```
Comment (gaius-qi) [inline internal/dynconfig/dynconfig.go:161]: Remove `Get` func.

File: internal/dynconfig/dynconfig_local.go [hunk 9]
```diff
@@ -45,3 +45,7 @@ func (d *dynconfigLocal) Unmarshal(rawVal interface{}) error {
 
 	return yaml.Unmarshal(b, rawVal)
 }
+
+func (d *dynconfigLocal) Get() (interface{}, error) {
```
Comment (gaius-qi) [inline internal/dynconfig/dynconfig_local.go:49]: Remove `Get` func

File: internal/dynconfig/dynconfig_manager.go [hunk 10]
```diff
@@ -82,6 +82,10 @@ func (d *dynconfigManager) Unmarshal(rawVal interface{}) error {
 	return decode(dynconfig, defaultDecoderConfig(rawVal))
 }
 
+func (d *dynconfigManager) Get() (interface{}, error) {
```
Comment (gaius-qi) [inline internal/dynconfig/dynconfig_manager.go:85]: Remove `Get` func.

File: internal/dynconfig/dynconfig.go [hunk 11]
```diff
@@ -156,6 +158,17 @@ func (d *Dynconfig) Unmarshal(rawVal interface{}) error {
...
+	return d.strategy.Get()
+}
+
+func (d *Dynconfig) Clean() error {
```
Comment (gaius-qi) [inline internal/dynconfig/dynconfig.go:165]: Add test for `Clean` func.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: internal/dynconfig/dynconfig.go (hunk 8 above)
Comment (244372610) [reply internal/dynconfig/dynconfig.go:161]: cdn dynamicConfig relies on this method

File: internal/dynconfig/dynconfig_local.go (hunk 9 above)
Comment (244372610) [reply internal/dynconfig/dynconfig_local.go:49]: same as above

File: internal/dynconfig/dynconfig.go (hunk 8 above)
Comment (gaius-qi) [reply internal/dynconfig/dynconfig.go:161]: Use `Unmarshal`.

**Pull request closed without merging**